river/preprocessing/one_hot.py

```diff
@@ -152,71 +113,23 @@ class OneHotListEncoder(OneHotEncoder):
     {'c1_a': 1, 'c1_b': 1, 'c2_x': 1}
     {'c1_i': 1, 'c2_h': 1, 'c2_z': 1}
     {'c1_b': 1, 'c1_h': 1, 'c2_e': 1}
-
-    The `sparse` parameter can be set to `False` in order to include the values that are not
-    present in the output.
-
-    >>> oh = river.preprocessing.OneHotListEncoder(sparse=False)
-    >>> for x in X[:2]:
-    ...     oh = oh.learn_one(x)
-    ...     pprint(oh.transform_one(x))
-    {'c1_a': 1, 'c1_u': 1, 'c2_d': 1}
-    {'c1_a': 1, 'c1_b': 1, 'c1_u': 0, 'c2_d': 0, 'c2_x': 1}
-
-    You can preserve the `c2` feature by using a union:
-
-    >>> from river import compose
-
-    >>> pp = compose.Select('c1') | river.preprocessing.OneHotListEncoder()
-    >>> pp += compose.Select('c2')
-
-    >>> for x in X:
-    ...     pp = pp.learn_one(x)
-    ...     pprint(pp.transform_one(x))
-    {'c1_a': 1, 'c1_u': 1, 'c2': ['d']}
-    {'c1_a': 1, 'c1_b': 1, 'c1_u': 0, 'c2': ['x']}
-    {'c1_a': 0, 'c1_b': 0, 'c1_i': 1, 'c1_u': 0, 'c2': ['h', 'z']}
-    {'c1_a': 0, 'c1_b': 1, 'c1_h': 1, 'c1_i': 0, 'c1_u': 0, 'c2': ['e']}
-
-    If you accidentally pass non list/set values to the encoder, it will not one-hot encode
-     the values In the example below, see the 'c2' feature and the one-hot encoded
-     outputs without this feature included.
-
-    >>> X = [{'c1': ['u', 'a'], 'c2': 'd'},
-    ...     {'c1': ['a', 'b'], 'c2': 'x'},
-    ...     {'c1': ['i'], 'c2': 'zh'},
-    ...     {'c1': ['h', 'b'], 'c2': 'e'}]
-    >>> oh = river.preprocessing.OneHotListEncoder(sparse=True)
-    >>> for x in X:
-    ...     oh = oh.learn_one(x)
-    ...     pprint(oh.transform_one(x))
-    {'c1_a': 1, 'c1_u': 1}
-    {'c1_a': 1, 'c1_b': 1}
-    {'c1_i': 1}
-    {'c1_b': 1, 'c1_h': 1}
     """
+
     def __init__(self, sparse=False):
-        super().__init__(sparse=sparse)
+        self.sparse = sparse
+        self.values = collections.defaultdict(set)
 
-    def learn_one(self, x: Dict[str, Union[set, list]]):
+    def learn_one(self, x):
         for i, xi in x.items():
-
-            # Similar to the OneHotEncoder class, but now
-            # xi is a list or a set
-            # If this is not the case, ignore
-            if type(xi) not in [list, set]:
-                continue
-
-            # Here, it's at least a list or a set.
-            # If it's empty, we don't have any elements and will
-            # effectively skip it. Otherwise, add all the elements of the
-            # list to the defaultdict
-            for xj in xi:
-                self.values[i].add(xj)
+            if isinstance(xi, (list, set, tuple)):
```

Comment:
I wouldn't include tuples. Indeed, those can be singular values in some case.

Reply:
Okay, I removed the tuple.